src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -654,6 +632,108 @@ public synchronized void updateEntity(DbEntity entity) throws DatabaseEngineExce
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+
+        dropFks(entity.getName(), fksToDrop);
+        if (!columnsToRemove.isEmpty()) {
+            if (properties.allowColumnDrop()) {
+                dropColumn(entity, columnsToRemove.toArray(new String[0]));
+            } else if (logger.isWarnEnabled()) {
+                logger.warn("Need to remove {} columns to update {} entity, but property allowColumnDrop is set to false.",
+                            StringUtils.join(columnsToRemove, ","), entity.getName()
+                );
+            }
+        }
+
+        List<DbColumn> columnsToAdd = new LinkedList<>();
+        for (DbColumn localColumn : entity.getColumns()) {
+            if (!currentEntityMetadata.containsKey(localColumn.getName())) {
+                columnsToAdd.add(localColumn);
+            }
+        }
+
+        if (!columnsToAdd.isEmpty()) {
+            addColumn(entity, columnsToAdd.toArray(new DbColumn[columnsToAdd.size()]));
+        }
+
+        Set<DbFk> fksToAdd = entity.getFks().stream()
+                                   .filter(fk -> !currentFks.containsValue(fk))
+                                   .collect(Collectors.toSet());
+
+        addFks(entity, fksToAdd);
+    }
+
+    /**
+     * Gets the current foreign keys in the table.
+     *
+     * <p>
+     * The entity metadata will be mapped to a {@link DbFk}.
+     * </p>
+     * <p>
+     * The resulting map will be the foreign key constraint name mapped to the {@link DbFk}.
+     * </p>
+     *
+     * @param entityName The entity name.
+     *
+     * @return Mapping between the foreign key constraint name and the {@link DbFk}.
+     *
+     * @throws DatabaseEngineException If it can't get the foreign keys metadata.
+     */
+    private Map<String, DbFk> getCurrentFks(final String entityName) throws DatabaseEngineException {
+        final Map<String, DbFk.Builder> existentFks = new HashMap<>();
+
+        try {
+            getConnection();
+            ResultSet rs = getMetadata().getImportedKeys(null, this.currentSchema, entityName);
+            while (rs.next()) {
+                final String fkName = rs.getString("FK_NAME");
+                final String foreignTable = rs.getString("FKTABLE_NAME");
+                final String localColumnName = rs.getString("PKCOLUMN_NAME");
+                final String foreignColumnName = rs.getString("FKCOLUMN_NAME");
+
+                existentFks.compute(fkName, (name, fk) -> fk == null
+                        ? new DbFk.Builder()
+                        .foreignTable(foreignTable)
+                        .addColumn(localColumnName)
+                        .addForeignColumn(foreignColumnName)
+                        : fk.foreignTable(foreignTable)
+                            .addColumn(localColumnName)
+                            .addForeignColumn(foreignColumnName)
+                );
```

Comment:
confusing...
Considering that a given `fkName` can only have 1 "foreignTable",
and assuming that  #201 is landed, you have rebased, and the proposed changes from the comment above have been accepted,
how about this?
```suggestion
                existentFks.computeIfAbsent(fkName, name -> new DbFk.Builder().referencedTable(referencedTable))
                        .addColumn(localColumnName)
                        .addReferencedColumn(referencedColumnName);
```